pkg/tekton/pipelines.go

```diff
@@ -118,13 +120,56 @@ func CreateOrUpdateTask(tektonClient tektonclient.Interface, ns string, created
 }
 
 // GenerateNextBuildNumber generates a new build number for the given project.
-func GenerateNextBuildNumber(tektonClient tektonclient.Interface, jxClient jxClient.Interface, ns string, gitInfo *gits.GitRepository, branch string, duration time.Duration, context string) (string, error) {
+func GenerateNextBuildNumber(tektonClient tektonclient.Interface, jxClient jxClient.Interface, ns string, gitInfo *gits.GitRepository, branch string, duration time.Duration, context string, useActivity bool) (string, error) {
 	nextBuildNumber := ""
 	resourceInterface := jxClient.JenkinsV1().SourceRepositories(ns)
+	activityInterface := jxClient.JenkinsV1().PipelineActivities(ns)
 	// TODO: How does SourceRepository handle name overlap?
 	sourceRepoName := naming.ToValidName(gitInfo.Organisation + "-" + gitInfo.Name)
 
 	f := func() error {
+		if useActivity {
```

Comment:
What's about splitting this function? Given that the two code paths are quite different I am wondering whether it even makes sense to have a BuildNumberGenerator interface for which you instantiate the right type depending on the flag. This would need some more thought probably. The idea would be to have a better separation of concerns and less entangled code.

Reply:
I can split this at the least - let's table the interface approach for later thought.

Reply:
ok